### PR TITLE
Add com.spotify.EdgeClient 

### PR DESCRIPTION
**Pull request by YuriBrandi**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [-] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** 

I have not contacted upsteam, however a non-beta version in already available, this is a 1:1 copy of it, but with auto-merging and bleeding edge upsteam.

- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by YuriBrandi** [comment]
bot, build com.spotify.EdgeClient

File: com.spotify.EdgeClient.json [hunk 1]
```diff
@@ -0,0 +1,276 @@
+{
+    "app-id": "com.spotify.EdgeClient",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "22.08",
```
Comment (hfiguiere) [inline com.spotify.EdgeClient.json:4]:
```suggestion
    "runtime-version": "23.08",
```

**Comment by YuriBrandi** [comment]
bot, build com.spotify.EdgeClient

**Comment by YuriBrandi** [comment]
bot, build com.spotify.EdgeClient

**Comment by YuriBrandi** [comment]
bot, build com.spotify.EdgeClient

**Comment by YuriBrandi** [comment]
bot, build com.spotify.EdgeClient

File: com.spotify.EdgeClient.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.spotify.EdgeClient</id>
+  <provides><id>com.spotify.EdgeClient.desktop</id></provides>
+  <launchable type="desktop-id">com.spotify.EdgeClient.desktop</launchable>
+  <name>Spotify</name>
+  <developer_name>Spotify</developer_name>
+  <summary>Online music streaming service</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary=https://www.spotify.com/us/legal/end-user-agreement/</project_license>
```
Comment (hfiguiere) [inline com.spotify.EdgeClient.appdata.xml:10]: is redistribution allowed?

File: com.spotify.EdgeClient.json (hunk 1 above)
Comment (hfiguiere) [reply com.spotify.EdgeClient.json:4]: this question still applies. Using obsolete runtime always lead to have them stay that way for longer than necessary.

File: com.spotify.EdgeClient.json [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+                    "type": "file",
+                    "path": "set-dark-theme-variant.py"
+                },
+                {
+                    "type": "extra-data",
+                    "filename": "spotify.snap",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://api.snapcraft.io/api/v1/snaps/download/pOBIoZ2LrCB3rDohMxoYGnbN14EHOgD7_69.snap",
```
Comment (hfiguiere) [inline com.spotify.EdgeClient.json:264]: what's the difference with https://github.com/flathub/com.spotify.Client/

Comment (YuriBrandi) [reply com.spotify.EdgeClient.json:264]: @hfiguiere This is the same Snap repackage but the upstream is bleeding edge and not stable. Should be realeased as a "beta" option for the same app.

File: com.spotify.EdgeClient.appdata.xml (hunk 2 above)
Comment (hfiguiere) [reply com.spotify.EdgeClient.appdata.xml:10]: that doesn't answer the question, it's just you speculating.

Comment (YuriBrandi) [reply com.spotify.EdgeClient.appdata.xml:10]: @hfiguiere I agree with you, I can contact Spotify and hope for an answer if it is preferred. However, not to be critical, but if this app can´t belong to Flathub then also https://github.com/flathub/com.spotify.Client/ should not.

File: com.spotify.EdgeClient.json (hunk 3 above)
Comment (hfiguiere) [reply com.spotify.EdgeClient.json:264]: two wrong don't make a right.

Comment (YuriBrandi) [reply com.spotify.EdgeClient.json:264]: > two wrong don't make a right.

So a beta version of an app is not allowed? Shall I close my pull request?

Comment (hfiguiere) [reply com.spotify.EdgeClient.json:264]: why not using the `beta` branch of the original package?

Comment (YuriBrandi) [reply com.spotify.EdgeClient.json:264]: It doesn't have one, so should I fork that repo and create a beta branch by myself and make a pull request?

**Comment by YuriBrandi** [comment]
> What exactly this is supposed to solve? We have allowed some beta versions in the regular Flathub as submitters could prove their are useful in some scenarios. If this is just another third-party maintained proprietary app but unstable, I don't think it belongs here.

The beta version usually goes stable after ten days more or less. The maintainer of the stable repo also has to manually accept flathubbot's PRs, which usually takes another week or so. I just wanted to offer users a bleeding edge version with automatically accepted PR. But I get creating a separate repo for a beta version is not the preferred approach here. So shall I close?

File: flathub.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "only-arches": ["x86_64"],
+  "automerge-flathubbot-prs": true
```
Comment (bbhtt) [inline flathub.json:3]: Automerging is a bad idea, imo .You aren't testing if anything breaks or regressions happened. Especially for something like a repackaged binary.

Someone has to deal with the fallout if a 'bad' build gets pushed to the users.

Comment (YuriBrandi) [reply flathub.json:3]: > Automerging is a bad idea, imo .You aren't testing if anything breaks or regressions happened. Especially for something like a repackaged binary.
> 
> Someone has to deal with the fallout if a 'bad' build gets pushed to the users. Are there any technical reasons for using it?

Thanks for the advice. Yes it was a temporary choice. However this was not meant to be a stable release anyway, also this Snap repackage breaks very rarely. 
I'm pretty sure this Flatpak won't make it to the public anyway, I'm going to close my pull request in a couple days.

**Pull request closed without merging**